### PR TITLE
Batch telemetry pings and send once per hour.

### PR DESCRIPTION
**Pull request by Osmose**
This also sends the pings on shutdown so that we can avoid having to store any persistent state between sessions.

### CODE REVIEW

**Comment by Osmose** [comment]
FYI the easiest way to test this is to go to `about:preferences` and select the search settings section; this logs an error to the console and is considered an error we collect by the add-on.

File: extension/ErrorReporting.jsm [hunk 1]
```diff
@@ -6,83 +6,85 @@ const {utils: Cu, interfaces: Ci, classes: Cc} = Components;
...
+    if (fingerprintsToReport.length > 0) {
+      const payload = {
+        version: 3,
+        study_name: "shield-study-js-errors",
+        branch: "default",
+        addon_version: "0.1.0",
+        shield_version: "unset",
+        type: "shield-study-addon",
+        data: {
+          fingerprints: fingerprintsToReport,
```
Comment (mythmon) [inline extension/ErrorReporting.jsm]: I think that the shield-study-addon schema requires all keys to be strings. You may need to `JSON.stringify()` this, which has worked fine on previous studies.

**Comment by Osmose** [comment]
Updated to better conform to the telemetry ping schemas.

**Review by mythmon — APPROVED** [review]

**Comment by Osmose** [comment]
Thanks!

**Pull request merged**